sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -810,6 +784,42 @@ mod checked {
                 written_objects.insert(id, object);
             }
 
+            // Before finishing, ensure that any shared object taken by value by the transaction is either:
+            // 1. Mutated (and still has a shared ownership); or
+            // 2. Deleted.
+            // Otherwise, the shared object operation is not allowed and we fail the transaction.
```

Comment:
Is the case that the object was re-shared but not deleted covered by mutation? Are we enforcing that the function must explicitly mutate the object or does the version increment cause the object to marked as mutated?

Reply:
It is! The version increment will always cause the object to be marked as mutated here.